DuckDuckGo/UserBehaviorMonitor.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  UserBehaviorMonitor.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Common
+import Core
+
+final class UserBehaviorMonitor {
+
+    enum Action: Equatable {
+
+        case refresh
+        case burn
+        case reopenApp
+        case openSettings
+        case toggleProtections
+
+    }
+
+    private let eventMapping: EventMapping<UserBehaviorEvent>
+    init(eventMapping: EventMapping<UserBehaviorEvent> = AppUserBehaviorMonitor.eventMapping) {
+        self.eventMapping = eventMapping
+    }
+
+    @UserDefaultsWrapper(key: .didRefreshTimestamp, defaultValue: .distantPast)
+    private var didRefreshTimestamp: Date?
+    @UserDefaultsWrapper(key: .didBurnTimestamp, defaultValue: .distantPast)
+    private var didBurnTimestamp: Date?
+
+    func handleAction(_ action: Action, date: Date = Date()) {
+        switch action {
+        case .refresh:
+            fireEventIfActionOccurredRecently(since: didRefreshTimestamp, eventToFire: .reloadTwice, within: 10.0)
```

Comment:
Do you think this also needs to take into account whether the reload has occurred on the same tab? This pixel can fire if you reload one tab, swap to another one, and then also reload that, which is likely not indicative of the sort of problem we're looking for.

Reply:
Marking resolved since this was discussed internally already.